### DISCUSSION TITLE
_neural_evolutionary_agent not learning

### DISCUSSION THREAD

**Discussion by HermesMG** [PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook#48] [Reinforcement Learning]
Hello:
I am executing the "_neural_evolutionary_agent,ipynb" notebook _first chapter- using 10 epochs and many times the agent does not learn, that is the reward is 0 for each step, and does not change.
In a few ocassions the reward changes with the epoch.
Can anybody explains me why I observe such strange results?
Thank you in advance
Hermes

**Comment by praveen-palanisamy** [comment]
Hi @HermesMG ,
For the [neural evolutionary agent](https://github.com/PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook/blob/master/Chapter01/06_neural_evolutionary_agent.py), `10` epochs is a very low number to train for convergence. In general for (approximate) Cross Entropy Methods especially with a neural network, the number of epochs needed is quite larger depending on the problem and the neural network's parameter count.
It is common to run for 200k+ epochs but if you are training with the default Brain and Agent definition and for the `Gridworld-v0` environment, you might see convergence with ~100k epochs or less. 
If training for that long is not feasible on your end, please try with a minimum of `1000-5000` epochs to see enough variance in the rewards.
